meson.build

```diff
@@ -45,6 +45,63 @@ if get_option('enable_debugger') == 'heavy'
   conf_data.set10('C_HEAVY_DEBUG', true)
 endif
 
+# prepare configuration for dynrec core
+#
+# host_machine is a machine we're building for
+#
+# In code we have code paths for two additional architectures:
+#
+# - ARMV4LE - autoconf used to set it when "armv6l" was reported, presumably
+#   it covers everything up to ARMv7.  Notably, Raspberry Pi 1 is using ARMv6.
+#   This arch was marked as NOT handling unaligned memory.
+# - MIPSEL - which used to be called MIPSEL32, and seems to be never hooked
+#   into the autoconf buildsystem (but it was used by several DOSBox ports)
+#
+selected_core = get_option('dynamic_core')
+host_cpu = host_machine.cpu_family()
+conf_data.set('C_TARGETCPU', 'UNKNOWN')
+cpu_handles_unaligned_memory = false
+
+core_selection = [
+  # cpu         selected_core       dynrec_define     C_TARGETCPU
+  [ 'x86_64',  ['auto', 'dyn-x86'], 'C_DYNAMIC_X86',  'X86_64'    ],
+  [ 'x86',     ['auto', 'dyn-x86'], 'C_DYNAMIC_X86',  'X86'       ],
+  [ 'x86_64',  ['dynrec'],          'C_DYNREC',       'X86_64'    ],
+  [ 'x86',     ['dynrec'],          'C_DYNREC',       'X86'       ],
+  [ 'aarch64', ['auto', 'dynrec'],  'C_DYNREC',       'ARMV8LE'   ],
+  [ 'arm',     ['auto', 'dynrec'],  'C_DYNREC',       'ARMV7LE'   ],
+  [ 'ppc64',   ['auto', 'dynrec'],  'C_DYNREC',       'PPC64LE'   ], # stable name
+  [ 'ppc64le', ['auto', 'dynrec'],  'C_DYNREC',       'PPC64LE'   ], # unstable name
+  [ 'ppc',     ['auto', 'dynrec'],  'C_DYNREC',       'POWERPC'   ],
+]
+
+foreach line : core_selection 
+  cpu = line[0]
+  opts_for_arch = line[1]
+  dynrec_define = line[2]
+  target_cpu = line[3]
+
+  if (host_cpu == cpu) and opts_for_arch.contains(selected_core)
+    conf_data.set('C_TARGETCPU', target_cpu)
+    conf_data.set(dynrec_define, 1)
+    cpu_handles_unaligned_memory = true
+  endif
+endforeach
```

Comment:
The foreach crawl the `core_selection` list looking for match. 
When a match is found (`host_cpu == cpu` and maching `arch`), should we break-out of the for-loop?

Reply:
Initially, I had `break` in here, but the keyword was added in Meson 0.49.0, so it's unavailable in meson from Ubuntu 18.04 repo.

When I tried to disregard this problem and use meson installed via pip3, it resulted in problems for builds in `platforms` workflow :( It really is much easier to stick to tools available via official repositories.

For now, I decided to skip breaking out of the loop in here. I think I'll ask Meson maintainer (who is also packager for Ubuntu) to backport newer meson version to 18.04.